blog/2023-03-01-recommended-architecture-for-a-simple-payment-dapp.md

```diff
@@ -0,0 +1,107 @@
+
+<!-- - Code examples: -->
+
+## Vanilla 🍦
+
+As this is very minimal dapp, this only supports Metamask out of the box, this is where librairies and framework come in handy. 
+  - _For instance, to support many more wallets without them, you would have to get into the documentation (if it exists) of each wallet and do a custom connector for each of them._
+The example focuses on two buttons: `Connect wallet` and `Transfer 0.1 CELO`. Let's go over each one.
+
+### Connecting a wallet
+
+To not clutter the codebase, we're using the `@metamask/onboarding` package in order to check for the wallet's presence in the page. Once we're sure MetaMask is installed, we use the `window.ethereum` API to directly request the wallet's first address via the following line: `await ethereum.request({ method: "eth_requestAccounts" });`. With it, we're now able to assess that the user's wallet is connected. To make it easier for ourselves, we're storing all potential important information in a `State` object.
+
+### Your first transaction
+
+Once connected, ~~we can now enable our transfer button.~~ we need to check if we have enough CELO for our first transaction. While this _sounds_ trivial, you actually need to know a couple things beforehand. First thing to note, `CELO` is a ERC20 tokens, and as such you're going to need to [interact with its smart-contract](https://developers.celo.org/17-powerful-celo-protocol-core-contracts-you-need-to-know-d84c1fbc5a6). To make it easy, we've already grabbed its ABI from [CELO's explorer](https://explorer.celo.org/), and saved it as `abi.json`.
+
+We recommended earlier using `ethers`, and that's just what we're going to do. Not that it's impossible to interact directly with RPC nodes and contracts, but this is totally outside the scope of this exercise.
+
+```js
+// This is the MetaMask provider (ethereum is injected by the browser extension)
+const metaMaskProvider = new ethers.providers.Web3Provider(window.ethereum);
+// Boilerplate, a signer represent the user's account basically
+const signer = metaMaskProvider.getSigner();
+// This is the CELO smart-contract representation in our code
+const CELOContract = new ethers.Contract(
+    TESTNET.NATIVE_TOKEN, // 0xf194afdf50b03e69bd7d057c1aa9e10c9954e4c9
+    TESTNET.NATIVE_TOKEN_ABI, // the content of abi.json
+    signer
+  );
+```
+
+So now, we have this `CELOContract` object… great?! But how does this help us?
+
+You can use it to call smart-contract methods (written in solidity) directly in your Javascript code! Such as `CELOContract.balanceOf` which is what's gonna determine if our user can transfer CELOs or not. 
+
+```js
+// All 
+State.balance = await CELOContract.balanceOf(State.address);
+if (State.balance.lt(TESTNET.AMOUNT)) {
```

Comment:
first thought: shouldnt this be greater than?

Reply:
yeah oops copied from my code where im disabling the button if smaller :D)